arca/_runner.py

```diff
@@ -0,0 +1,67 @@
+""" This file is the code which actually launches the tasks, the serialized jsons.
+"""
+import json
+import traceback
+import sys
+import os
+from importlib import import_module
+from pathlib import Path
+
+sys.path.insert(1, os.getcwd())
+
+
+class EntryPoint:
+    def __init__(self, module_name, object_name):
+        self.module_name = module_name
+        self.object_name = object_name
+
+    def load(self):
+        mod = import_module(self.module_name)
+        obj = mod
+        for attr in self.object_name.split('.'):
+            obj = getattr(obj, attr)
+
+        return obj
+
+
+def run(filename):
+    try:
+
+        task_definition = json.loads(Path(filename).read_text())
+
+        entry_point = EntryPoint(task_definition["entry_point"]["module_name"],
+                                 task_definition["entry_point"]["object_name"])
+
+        args = task_definition["args"]
+        kwargs = task_definition["kwargs"]
+
+        if not isinstance(args, list):
+            raise ValueError("args is not a list")
+
+        if not isinstance(kwargs, dict) or not all([isinstance(x, str) for x in kwargs.keys()]):
+            raise ValueError("kwargs is not a valid kwargs dict")
+
+    except (ValueError, KeyError, TypeError, AttributeError):
+        # ValueError: the json can be corrupted, args and kwargs can be invalid
+        # KeyError: keys can be missing from the parsed json
+        # TypeError: task_definition["entry_point"] is a list
+        # AttributeError: task_definition["entry_point"] isn't a dict
+
+        return {"success": False, "reason": "corrupted_definition", "error": traceback.format_exc()}
+
+    try:
+        entry_point = entry_point.load()
+    except (ImportError, AttributeError):
+        return {"success": False, "reason": "import", "error": traceback.format_exc()}
+
+    res = entry_point(*args, **kwargs)
+
+    return {"success": True, "result": res}
+
+
+if __name__ == "__main__":
+    try:
+        # <script> <json>
+        print(json.dumps(run(sys.argv[1])))
+    except:  # noqa:
```

Comment:
Instead of noqa, try BaseException to indicate that you want everything. Bare except usually indicates you don't know or was lazy.

Note that if you have a syntax error or something like that inside run, this will catch it. while if you put the handling above, you are lowering those risks for less code.